### DISCUSSION TITLE
Can't create a valid SpanContext from data of another SpanContext

### DISCUSSION THREAD

**Discussion by Shlomo-Stein-LS** [open-telemetry/opentelemetry-cpp#1563] [Q&A] [answered]
Hi guys, I need your help,
I'm having trouble creating a new SpanContext obj from data I saved earlier that I got from another SpanContext.

Here is my code:

`// Creating a new span`
`nostd::shared_ptr<trace::Span> openSpan = GetTracer(name)->StartSpan(name);`

`// Getting data from context`
`trace::SpanContext context = openSpan ->GetContext();`
`const uint8_t* traceId = context.trace_id ().Id ().data ();                // ----> traceId contains garbage data`
`const uint8_t* spanId = context.span_id ().Id ().data ();                // ----> spanId  contains garbage data`
`uint8_t flags = context.trace_flags().flags ();`
`// FYI, I'm passing that data to a different service`

`// Initializing a new Span context from the parameters`
`void createSpanContext (uint8_t* trace_id, uint8_t* span_id, uint8_t trace_flags)`
`{`
`     nostd::span<const uint8_t, trace::TraceId::kSize> traceIdData (trace_id, trace::TraceId::kSize);`
`     nostd::span<const uint8_t, trace::SpanId::kSize>  spanIdData  (span_id, trace::SpanId::kSize);`
         
`     trace::TraceId      traceId (traceIdData);`
`     trace::SpanId       spanId  (spanIdData);`
`     trace::TraceFlags   flags   (trace_flags);`
  
`     trace::SpanContext  newContext (traceId, spanId, flags, false);`
........
`}`

I can't figure out why I'm getting garbage data from the SpanContext
Please help, tell me what I'm doing wrong.

**Comment by owent** [comment, marked as answer]
What do you mean by saying "garbage data"? TraceID and SpanID are binary data and are not printable.The only invalid value is all zero.

**Reply by Shlomo-Stein-LS** [reply]
I figured out the issue, it seems like I did not parse the data correctly.
Thanks anyways
